src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Runtime.InteropServices;
+using System.IO;
+using System.Threading.Tasks;
+using System.Collections.Concurrent;
+
+namespace System.Device.Gpio.Drivers
+{
+    public class LibGpiodDriver : UnixDriver
+    {
+        private SafeChipHandle _chip;
+
+        private Dictionary<int, SafeLineHandle> _pinNumberToSafeLineHandle;
+
+        private ConcurrentDictionary<int, LibGpiodDriverEventHandler> _pinNumberToEventHandler = new ConcurrentDictionary<int, LibGpiodDriverEventHandler>();
+
+        protected internal override int PinCount => Interop.GetNumberOfLines(_chip);
+
+        public LibGpiodDriver()
+        {
+            OpenChip();
+            _pinNumberToSafeLineHandle = new Dictionary<int, SafeLineHandle>(PinCount);
+        }
+
+        private void OpenChip()
+        {
+            SafeChipIteratorHandle iterator = Interop.GetChipIterator();
+            if (iterator == null)
+            {
+                throw new IOException($"No any chip available, error code: {Marshal.GetLastWin32Error()}");
+            }
+            _chip = Interop.GetNextChipFromChipIterator(iterator);
+            if (_chip == null)
+            {
+                throw new IOException($"No chip found, error code: {Marshal.GetLastWin32Error()}");
+            }
+            // Freeing other chips opened
+            Interop.FreeChipIteratorNoCloseCurrentChip(iterator);
+        }
+
+        private SafeLineHandle RequestEvent(int pinNumber, PinEventTypes eventType)
+        {
+            int eventSuccess = -1;
+            SafeLineHandle pin = Interop.GetChipLineByOffset(_chip, pinNumber);
+            if (pin != null)
+            {
+                if (eventType == PinEventTypes.Rising || eventType == PinEventTypes.Falling)
```

Comment:
NIT: you can use the .HasFlags instead in here instead of comparing it. If you have both rising and falling then you won't get into the if because PinEventTypes is a flags enum. Does this mean that we will get an event for both rising and falling even if we only care about one of those?

Reply:
`you can use the .HasFlags instead in here instead of comparing it. `
it was Dan's suggestion https://github.com/dotnet/iot/pull/133#discussion_r245459095
`If you have both rising and falling then you won't get into the if because PinEventTypes is a flags enum` 
this point makes sense, i was not aware of that feature of flag enum, will consider that, thanks
`Does this mean that we will get an event for both rising and falling even if we only care about one of those?`
Yes, we will listen for both events and when event occurred will lookup if there is any handler/delegate registered for that event.